heat/core/_operations.py

```diff
@@ -62,194 +62,202 @@ def __binary_op(
     b = ht.zeros(10000, split=0)
     c = a[:-1] + b[1:]
     ```
-    In such cases, one of the operands is redistributed IN PLACE to match the distribution map of the other operand.
+    In such cases, one of the operands is redistributed OUT-OF-PLACE to match the distribution map of the other operand.
     """
+    # Check inputs
+    if not np.isscalar(t1) and not isinstance(t1, DNDarray):
+        raise TypeError(
+            "Only tensors and numeric scalars are supported, but input was {}".format(type(t1))
+        )
+    if not np.isscalar(t2) and not isinstance(t2, DNDarray):
+        raise TypeError(
+            "Only tensors and numeric scalars are supported, but input was {}".format(type(t2))
+        )
```

Comment:
At some point, we were thinking to allow interchangeable operations with PyTorch tensors / NumPy arrays. We can try to create DNDarrays out of the inputs here.